workers/datasets_based/src/datasets_based/worker.py

```diff
@@ -237,17 +244,26 @@ def should_skip_job(self) -> bool:
             cached_response = get_response_without_content(
                 kind=self.processing_step.cache_kind, dataset=self.dataset, config=self.config, split=self.split
             )
+        except DoesNotExist:
+            # no entry in the cache
+            return False
+        if cached_response["error_code"] in ERROR_CODES_TO_RETRY:
+            # the cache entry result was a temporary error - we process it
+            return False
+        if (
+            cached_response["worker_version"] is None
+            or self.compare_major_version(cached_response["worker_version"]) != 0
+        ):
+            # no worker version in the cache, or the worker has been updated - we process the job to update the cache
+            return False
+        try:
             dataset_git_revision = self.get_dataset_git_revision()
-            return (
-                # TODO: use "error_code" to decide if the job should be skipped (ex: retry if temporary error)
-                cached_response["http_status"] == HTTPStatus.OK
-                and cached_response["worker_version"] is not None
-                and self.compare_major_version(cached_response["worker_version"]) == 0
-                and cached_response["dataset_git_revision"] is not None
-                and cached_response["dataset_git_revision"] == dataset_git_revision
-            )
         except Exception:
+            # an exception occurred while getting the git revision from the Hub - the job will fail anyway, but we
+            # process it to store the error in the cache
             return False
+        return dataset_git_revision is not None and cached_response["dataset_git_revision"] == dataset_git_revision
```

Comment:
Can we also make sure to not **emit** jobs if this condition is met ? This could help making the queue less full of unnecessary jobs

Reply:
Yes, sure, I'd planned to do it for #736. But I'm trying to do small PRs, I think it will work better now that we work as a team

Reply:
nice !